torchok/constructor/constructor.py

```diff
@@ -281,22 +281,31 @@ def create_dataloaders(self, phase: Phase) -> List[DataLoader]:
             - ValueError: When OneOrOther composition is passed that isn't supported
         """
         if phase in self.hparams.data:
-            dataloaders = [
-                self._prepare_dataloader(phase_params.dataset, phase_params.dataloader)
-                for phase_params in self.hparams.data[phase] if phase_params is not None
-            ]
+            dataloaders = []
+            for phase_params in self.hparams.data[phase]:
+                if phase_params is not None:
+                    sampler_params = phase_params.sampler if phase_params.get('sampler') else None
```

Comment:
you can write `phase_params.get('sampler')` if 'sampler not in dict it return None